Add dags only deploy template for private network (#3210)
astronomer/docs: astro/ci-cd-templates/github-actions.md

If you use GitHub Enterprise and can't use the public Astronomer [Deploy Action](https://github.com/astronomer/deploy-action) in the GitHub Marketplace, use the following templates to implement CI/CD.

> You can configure your CI/CD pipelines to deploy a full image or

Reviewer: ```suggestion
You can configure your CI/CD pipelines to deploy a full project image or your `dags` directory.
```

> ### Full image deploy templates

Reviewer: ```suggestion
### Project deploy templates
```
I think people could mistake the original header as meaning "you can't do DAG deploys with these"

Author: Oops this is an artifact of the original structure I'd made with full image + dags only in separate sections. I don't think we need it anymore.

<Tabs
    defaultValue="standard"

>     groupId= "private-network-templates"

Reviewer: Whatever the title is above, we should change this group ID to be the same as the title's anchor link

   - `ASTRO_API_TOKEN`: The value for your Workspace or Organization API token.

2. In your project repository, create a new YAML file in `.github/workflows` that includes the following configuration. When you make a commit to a specified branch, this workflow sets your Deployment API credentials as environment variables, installs the latest version of the Astro CLI, checks to see if your `dags` folder has changes, and then either completes a full code deploy or a DAG-only code deploy.

>     ```yaml

Reviewer: Something's not right with how this renders in the preview, it shows the `yaml` line as part of the code block. Indenting by two spaces might work? But otherwise it's probably fine to un-indent the whole thing

Author: for some reason it was the two blank lines!

Reviewer: Ah ok that did the trick!

Reviewer: For indented content, Markdown will "forget" that it's within a list if there's more than one new line. Because of the two lines here, it exited the list format and instead used the fallback for indented formatting, which is to make a code fence

   - `ASTRO_API_TOKEN`: The value for your Workspace or Organization API token.

> 2. In your project repository, create a new YAML file in `.github/workflows` that includes the following configuration. the following template sets your Deployment API credentials as environment variables, makes sure it uses the most recent Astro CLI version, checks to see if only DAG files have changes, and then either completes a full code deploy or a DAG-only code deploy.

Reviewer: ```suggestion
2. In your project repository, create a new YAML file in `.github/workflows` that includes the following configuration. When you make a commit to a specified branch, this workflow sets your Deployment API credentials as environment variables, installs the latest version of the Astro CLI, checks to see if your `dags` folder has changes, and then either completes a full code deploy or a DAG-only code deploy.
```

<TabItem value="custom">

> If your Astro project requires additional build-time arguments to build an image, you need to define these build arguments using Docker's [`build-push-action`](https://github.com/docker/build-push-action). This template pushes a [full image deploy](deploy-project-image.md) to Astro.

Reviewer: ```suggestion
If your Astro project requires additional build-time arguments to build an image, you need to define these build arguments using Docker's [`build-push-action`](https://github.com/docker/build-push-action). This template always pushes your entire project [as an image](deploy-project-image.md) to Astro.
```
